Index: service/hibernate/src/hibernate/org/sakaiproject/tool/gradebook/Category.hbm.xml
===================================================================
--- service/hibernate/src/hibernate/org/sakaiproject/tool/gradebook/Category.hbm.xml	(revision 64836)
+++ service/hibernate/src/hibernate/org/sakaiproject/tool/gradebook/Category.hbm.xml	(working copy)
@@ -24,6 +24,17 @@
 		<property name="drop_lowest" column="DROP_LOWEST" type="integer" not-null="false" />
 		
 		<property name="removed" column="REMOVED" type="boolean" />
+		
+		<property name="extraCredit" column="IS_EXTRA_CREDIT" type="boolean" not-null="false" />
+ 		
+	 	<property name="equalWeightAssignments" column="IS_EQUAL_WEIGHT_ASSNS" type="boolean" not-null="false" />
+	
+		<property name="unweighted" column="IS_UNWEIGHTED" type="boolean" not-null="false" />
+	
+		<property name="categoryOrder" column="CATEGORY_ORDER" type="integer" not-null="false" />
+	
+		<property name="enforcePointWeighting" column="ENFORCE_POINT_WEIGHTING" type="boolean" not-null="false" />
+			
   </class>
   
  	<database-object>
Index: service/hibernate/src/hibernate/org/sakaiproject/tool/gradebook/Gradebook.hbm.xml
===================================================================
--- service/hibernate/src/hibernate/org/sakaiproject/tool/gradebook/Gradebook.hbm.xml	(revision 64836)
+++ service/hibernate/src/hibernate/org/sakaiproject/tool/gradebook/Gradebook.hbm.xml	(working copy)
@@ -36,5 +36,20 @@
 		
 		<property name="category_type" column="CATEGORY_TYPE" type="integer" not-null="true"/>
 
+		<property name="equalWeightCategories" column="IS_EQUAL_WEIGHT_CATS" type="boolean" not-null="false"/>
+
+		<property name="scaledExtraCredit" column="IS_SCALED_EXTRA_CREDIT" type="boolean" not-null="false"/>
+
+		<property name="showMean" column="DO_SHOW_MEAN" type="boolean" not-null="false"/>
+		
+		<property name="showMedian" column="DO_SHOW_MEDIAN" type="boolean" not-null="false"/>
+		
+		<property name="showMode" column="DO_SHOW_MODE" type="boolean" not-null="false"/>
+		
+		<property name="showRank" column="DO_SHOW_RANK" type="boolean" not-null="false"/>
+		
+		<property name="showItemStatistics" column="DO_SHOW_ITEM_STATS" type="boolean" not-null="false"/>
+		
+
 	</class>
 </hibernate-mapping>
Index: service/hibernate/src/hibernate/org/sakaiproject/tool/gradebook/GradableObject.hbm.xml
===================================================================
--- service/hibernate/src/hibernate/org/sakaiproject/tool/gradebook/GradableObject.hbm.xml	(revision 64836)
+++ service/hibernate/src/hibernate/org/sakaiproject/tool/gradebook/GradableObject.hbm.xml	(working copy)
@@ -21,6 +21,8 @@
 
 		<property name="removed" column="REMOVED" type="boolean" />
 
+		<property name="sortOrder" column="SORT_ORDER" type="java.lang.Integer" />
+
 		<subclass name="org.sakaiproject.tool.gradebook.Assignment" extends="org.sakaiproject.tool.gradebook.GradableObject"  discriminator-value="1">
 			<property name="pointsPossible" column="POINTS_POSSIBLE" type="double" />
 			<property name="dueDate" column="DUE_DATE" type="date" />
@@ -33,6 +35,7 @@
       <property name="released" column="RELEASED" type="boolean"/>
       <many-to-one name="category" column="CATEGORY_ID" class="org.sakaiproject.tool.gradebook.Category" not-null="false"/>
       <property name="ungraded" column="UNGRADED" type="boolean"/>
+      <property name="countNullsAsZeros" column="IS_NULL_ZERO" type="boolean" not-null="false"/>
     </subclass>
 
 		<subclass name="org.sakaiproject.tool.gradebook.CourseGrade" extends="org.sakaiproject.tool.gradebook.GradableObject" discriminator-value="2">
Index: service/hibernate/src/java/org/sakaiproject/tool/gradebook/CourseGradeRecord.java
===================================================================
--- service/hibernate/src/java/org/sakaiproject/tool/gradebook/CourseGradeRecord.java	(revision 64836)
+++ service/hibernate/src/java/org/sakaiproject/tool/gradebook/CourseGradeRecord.java	(working copy)
@@ -38,7 +38,7 @@
 public class CourseGradeRecord extends AbstractGradeRecord {
     private String enteredGrade;
     private Double autoCalculatedGrade;  // Not persisted
-    private Double calculatedPointsEarned;	// Not persisted
+	private Double calculatedPointsEarned;	// Not persisted
 
     public static Comparator<CourseGradeRecord> calcComparator;
 
@@ -215,4 +215,10 @@
 		}
 		autoCalculatedGrade = percentageEarned;
 	}
+	public Double getCalculatedPointsEarned() {
+		return calculatedPointsEarned;
+	}
+	public void setAutoCalculatedGrade(Double autoCalculatedGrade) {
+		this.autoCalculatedGrade = autoCalculatedGrade;
+	}
 }
Index: service/hibernate/src/java/org/sakaiproject/tool/gradebook/Assignment.java
===================================================================
--- service/hibernate/src/java/org/sakaiproject/tool/gradebook/Assignment.java	(revision 64836)
+++ service/hibernate/src/java/org/sakaiproject/tool/gradebook/Assignment.java	(working copy)
@@ -67,6 +67,9 @@
     private Category category;
     private Double averageTotal;
     private boolean ungraded;
+    private Boolean extraCredit;
+  	private Double assignmentWeighting;
+    private Boolean countNullsAsZeros;
 
     static {
         dateComparator = new Comparator() {
@@ -434,7 +437,11 @@
 			this.averageTotal = averageTotal;
 		}
 
-
+		public boolean getUngraded()
+		{
+			return ungraded;
+		}
+		
 		public boolean isUngraded()
 		{
 			return ungraded;
@@ -444,4 +451,33 @@
 		{
 			this.ungraded = ungraded;
 		}
+
+		public Boolean getCountNullsAsZeros() {
+			return countNullsAsZeros;
+		}
+
+
+		public void setCountNullsAsZeros(Boolean countNullsAsZeros) {
+			this.countNullsAsZeros = countNullsAsZeros;
+		}
+
+
+		public Boolean isExtraCredit() {
+			return extraCredit;
+		}
+
+
+		public void setExtraCredit(Boolean extraCredit) {
+			this.extraCredit = extraCredit;
+		}
+
+
+		public Double getAssignmentWeighting() {
+			return assignmentWeighting;
+		}
+
+
+		public void setAssignmentWeighting(Double assignmentWeighting) {
+			this.assignmentWeighting = assignmentWeighting;
+		}
 }
\ No newline at end of file
Index: service/hibernate/src/java/org/sakaiproject/tool/gradebook/AssignmentGradeRecord.java
===================================================================
--- service/hibernate/src/java/org/sakaiproject/tool/gradebook/AssignmentGradeRecord.java	(revision 64836)
+++ service/hibernate/src/java/org/sakaiproject/tool/gradebook/AssignmentGradeRecord.java	(working copy)
@@ -38,6 +38,10 @@
     private String letterEarned;
     private Double percentEarned;
     private boolean userAbleToView;
+    private Boolean excludedFromGrade;
+    private transient BigDecimal earnedWeightedPercentage;
+    private transient BigDecimal overallWeight;
+    private transient Boolean isDropped;
 
     public AssignmentGradeRecord() {
         super();
@@ -165,6 +169,38 @@
     	agr.setStudentId(studentId);
     	return agr;
     }
+
+	public Boolean isExcludedFromGrade() {
+		return excludedFromGrade;
+	}
+
+	public void setExcludedFromGrade(Boolean excludedFromGrade) {
+		this.excludedFromGrade = excludedFromGrade;
+	}
+
+	public BigDecimal getEarnedWeightedPercentage() {
+		return earnedWeightedPercentage;
+	}
+
+	public void setEarnedWeightedPercentage(BigDecimal earnedWeightedPercentage) {
+		this.earnedWeightedPercentage = earnedWeightedPercentage;
+	}
+
+	public BigDecimal getOverallWeight() {
+		return overallWeight;
+	}
+
+	public void setOverallWeight(BigDecimal overallWeight) {
+		this.overallWeight = overallWeight;
+	}
+
+	public Boolean isDropped() {
+		return isDropped;
+	}
+
+	public void setDropped(Boolean isDropped) {
+		this.isDropped = isDropped;
+	}
 }
 
 
Index: service/hibernate/src/java/org/sakaiproject/tool/gradebook/Category.java
===================================================================
--- service/hibernate/src/java/org/sakaiproject/tool/gradebook/Category.java	(revision 64836)
+++ service/hibernate/src/java/org/sakaiproject/tool/gradebook/Category.java	(working copy)
@@ -21,6 +21,11 @@
 	private Double mean; //mean value of percentage for this category
 	private List assignmentList;
 	private int assignmentCount;
+	private Boolean extraCredit;
+	private Boolean unweighted;
+	private Boolean equalWeightAssignments;
+	private Integer categoryOrder;
+	private Boolean enforcePointWeighting;
 	
 	public static Comparator nameComparator;
 	public static Comparator averageScoreComparator;
@@ -342,4 +347,44 @@
 	public void setAssignmentCount(int assignmentCount){
 		this.assignmentCount = assignmentCount;
 	}
+	
+	public Boolean isUnweighted() {
+		return unweighted;
+	}
+
+	public void setUnweighted(Boolean unweighted) {
+		this.unweighted = unweighted;
+	}
+
+	public Boolean isEqualWeightAssignments() {
+		return equalWeightAssignments;
+	}
+
+	public void setEqualWeightAssignments(Boolean equalWeightAssignments) {
+		this.equalWeightAssignments = equalWeightAssignments;
+	}
+
+	public Integer getCategoryOrder() {
+		return categoryOrder;
+	}
+
+	public void setCategoryOrder(Integer categoryOrder) {
+		this.categoryOrder = categoryOrder;
+	}
+
+	public Boolean isEnforcePointWeighting() {
+		return enforcePointWeighting;
+	}
+
+	public void setEnforcePointWeighting(Boolean enforcePointWeighting) {
+		this.enforcePointWeighting = enforcePointWeighting;
+	}
+
+	public Boolean isExtraCredit() {
+		return extraCredit;
+	}
+
+	public void setExtraCredit(Boolean extraCredit) {
+		this.extraCredit = extraCredit;
+	}
 }
Index: service/hibernate/src/java/org/sakaiproject/tool/gradebook/GradableObject.java
===================================================================
--- service/hibernate/src/java/org/sakaiproject/tool/gradebook/GradableObject.java	(revision 64836)
+++ service/hibernate/src/java/org/sakaiproject/tool/gradebook/GradableObject.java	(working copy)
@@ -43,6 +43,7 @@
     protected int version;
     protected Gradebook gradebook;
     protected String name;
+    protected Integer sortOrder;
     protected Double mean;	// not persisted; not used in all contexts (in Overview & Assignment Grading,
     	                    // not in Roster or Student View)
 
@@ -51,7 +52,7 @@
                                 // delete the assignment.  Instead, we should hide the "removed" assignments
                                 // from the app by filtering the removed assignments in the hibernate queries
 
-
+    
     /**
      * @return Whether this gradable object is a course grade
      */
@@ -186,6 +187,14 @@
           append(name).
           toHashCode();
 	}
+
+	public Integer getSortOrder() {
+		return sortOrder;
+	}
+
+	public void setSortOrder(Integer sortOrder) {
+		this.sortOrder = sortOrder;
+	}
 }
 
 
Index: service/hibernate/src/java/org/sakaiproject/tool/gradebook/Gradebook.java
===================================================================
--- service/hibernate/src/java/org/sakaiproject/tool/gradebook/Gradebook.java	(revision 64836)
+++ service/hibernate/src/java/org/sakaiproject/tool/gradebook/Gradebook.java	(working copy)
@@ -49,6 +49,13 @@
     private boolean locked;
     private int grade_type;
     private int category_type;
+    private Boolean equalWeightCategories;
+    private Boolean scaledExtraCredit;
+    private Boolean showMean;
+    private Boolean showMedian;
+    private Boolean showMode;
+    private Boolean showRank;
+    private Boolean showItemStatistics;
 
     /**
      * Default no-arg constructor needed for persistence
@@ -252,6 +259,66 @@
 		{
 			this.grade_type = grade_type;
 		}
+
+		public Boolean isEqualWeightCategories() {
+			return equalWeightCategories;
+		}
+
+		public void setEqualWeightCategories(Boolean equalWeightCategories) {
+			this.equalWeightCategories = equalWeightCategories;
+		}
+
+		public Boolean isScaledExtraCredit() {
+			return scaledExtraCredit;
+		}
+
+		public void setScaledExtraCredit(Boolean scaledExtraCredit) {
+			this.scaledExtraCredit = scaledExtraCredit;
+		}
+
+		public Boolean getShowMean() {
+			return showMean;
+		}
+
+		public void setShowMean(Boolean showMean) {
+			this.showMean = showMean;
+		}
+
+		public Boolean getShowMedian() {
+			return showMedian;
+		}
+
+		public void setShowMedian(Boolean showMedian) {
+			this.showMedian = showMedian;
+		}
+
+		public Boolean getShowMode() {
+			return showMode;
+		}
+
+		public void setShowMode(Boolean showMode) {
+			this.showMode = showMode;
+		}
+
+		public Boolean getShowRank() {
+			return showRank;
+		}
+
+		public void setShowRank(Boolean showRank) {
+			this.showRank = showRank;
+		}
+
+		public Boolean getShowItemStatistics() {
+			return showItemStatistics;
+		}
+
+		public void setShowItemStatistics(Boolean showItemStatistics) {
+			this.showItemStatistics = showItemStatistics;
+		}
+
+		public Boolean getScaledExtraCredit() {
+			return scaledExtraCredit;
+		}
 }
 
 
